rosserial_server/include/rosserial_server/session.h

```diff
@@ -383,6 +407,11 @@ class Session : boost::noncopyable
     return (val >> 8) + val;
   }
 
+  static void signal_catch(int sig)
+  {
+    terminate_start_flag_ = true;
```

Comment:
If we're catching this, do we need to manually propagate it? Per http://wiki.ros.org/roscpp/Overview/Initialization%20and%20Shutdown, I think we need a `ros::shutdown();` here.

Reply:
Do you mean to move the following action inside this static function?

https://github.com/ros-drivers/rosserial/blob/acf73b7627f6910514165415dd6792f4727e0e20/rosserial_server/include/rosserial_server/session.h#L178-L181

Or do you mean that we do not need to send `rosserial_msgs::TopicInfo::ID_TX_STOP`, and only need to do `ros:shutdown()` for the stop action?